griptape/common/prompt_stack/messages/prompt_stack_message.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+from typing import Any
+
+from attrs import define, field
+
+from griptape.artifacts import TextArtifact
+from griptape.common import BasePromptStackContent, TextPromptStackContent
+
+from .base_prompt_stack_message import BasePromptStackMessage
+
+
+@define
+class PromptStackMessage(BasePromptStackMessage):
+    def __init__(self, content: str | list[BasePromptStackContent], **kwargs: Any):
+        if isinstance(content, str):
+            content = [TextPromptStackContent(TextArtifact(value=content))]
+        self.__attrs_init__(content, **kwargs)  # pyright: ignore[reportAttributeAccessIssue]
+
+    content: list[BasePromptStackContent] = field(metadata={"serializable": True})
```

Comment:
Went back and forth on this a million times. In this PR, yeah it's not much more. But the upcoming function calling PR, for instance, has an `ActionResultPromptStackContent` that contains an artifact _and_ some metadata about the action.

Additionally, to quote something @vasinov said:
> I think the reason we wanted to keep it separate is because artifacts are a pretty clean abstraction for data passing. In the PromptStack inputs really are elements that currently somewhat match artifacts but we don't know if that's going to hold true in the future. It would also be much easier to add API-specific elements if an API requires it.